source/common/config/filter_json.cc

```diff
@@ -214,5 +214,22 @@ void FilterJson::translateHttpConnectionManager(
   }
 }
 
+void FilterJson::translateMongoProxy(const Json::Object& json_mongo_proxy,
+                                     envoy::api::v2::filter::network::MongoProxy& mongo_proxy) {
+  json_mongo_proxy.validateSchema(Json::Schema::MONGO_PROXY_NETWORK_FILTER_SCHEMA);
+
+  JSON_UTIL_SET_STRING(json_mongo_proxy, mongo_proxy, stat_prefix);
+  JSON_UTIL_SET_STRING(json_mongo_proxy, mongo_proxy, access_log);
+  if (json_mongo_proxy.hasObject("fault")) {
+    const auto json_fault = json_mongo_proxy.getObject("fault")->getObject("fixed_delay");
+    auto* delay = mongo_proxy.mutable_delay();
+
+    delay->set_type(envoy::api::v2::filter::FaultDelay::FaultDelayType::FIXED);
+    delay->set_percent(static_cast<uint32_t>(json_fault->getInteger("percent")));
+    delay->mutable_fixed_delay()->CopyFrom(
+        Protobuf::util::TimeUtil::MillisecondsToDuration(json_fault->getInteger("duration_ms")));
```

Comment:
`JSON_UTIL_SET_DURATION`, we probably need a variant where the old/new field names are different as well.